src/pages/configurePatch/configurePatchCore/ConfigureTasks.tsx

```diff
@@ -39,40 +57,73 @@ export const ConfigureTasks: React.FC<Props> = ({
+  );
+
+  // Show details related to child patches (i.e. list all variants/tasks) only if it is the only menu item selected
+  const enumerateChildPatches =
+    currentChildPatches.length === 1 && selectedBuildVariants.length === 1;
 
   const onClickCheckbox = (taskName: string) => (e) => {
     const selectedBuildVariantsCopy = { ...selectedBuildVariantTasks };
-    selectedBuildVariants.forEach((bv) => {
-      if (selectedBuildVariantsCopy[bv][taskName] !== undefined) {
-        selectedBuildVariantsCopy[bv][taskName] = e.target.checked;
+    const selectedDownstreamPatchesCopy = { ...selectedDownstreamPatches };
+    selectedBuildVariants.forEach((v) => {
+      if (selectedBuildVariantsCopy?.[v]?.[taskName] !== undefined) {
+        selectedBuildVariantsCopy[v][taskName] = e.target.checked;
+      } else if (selectedDownstreamPatchesCopy?.[v] !== undefined) {
+        selectedDownstreamPatchesCopy[v] = e.target.checked;
       }
     });
     setSelectedBuildVariantTasks(selectedBuildVariantsCopy);
+    setSelectedDownstreamPatches(selectedDownstreamPatchesCopy);
   };
 
   const onClickSelectAll = (e) => {
     const selectedBuildVariantsCopy = { ...selectedBuildVariantTasks };
-    selectedBuildVariants.forEach((bv) => {
-      Object.keys(selectedBuildVariantsCopy[bv]).forEach((task) => {
-        selectedBuildVariantsCopy[bv][task] = e.target.checked;
-      });
+    const selectedDownstreamPatchesCopy = { ...selectedDownstreamPatches };
+    selectedBuildVariants.forEach((v) => {
+      if (selectedBuildVariantsCopy?.[v] !== undefined) {
+        Object.keys(selectedBuildVariantsCopy[v]).forEach((task) => {
+          selectedBuildVariantsCopy[v][task] = e.target.checked;
+        });
+      } else if (selectedDownstreamPatchesCopy?.[v] !== undefined) {
+        selectedDownstreamPatchesCopy[v] = e.target.checked;
+      }
     });
     setSelectedBuildVariantTasks(selectedBuildVariantsCopy);
+    setSelectedDownstreamPatches(selectedDownstreamPatchesCopy);
   };
 
-  const selectAllCheckboxState = getSelectAllCheckboxState(currentTasks);
-  const selectAllCheckboxCopy = `Select all tasks in ${
-    selectedBuildVariants.length > 1 ? "these variants" : "this variant"
-  }`;
+  const selectAllCheckboxState = getSelectAllCheckboxState(
+    currentTasks,
+    currentDownstreamPatches,
+    enumerateChildPatches
+  );
+  const selectAllCheckboxCopy =
+    Object.entries(currentTasks).length === 0
+      ? `Select all tasks in ${
+          selectedBuildVariants.length > 1
+            ? "these trigger aliases"
+            : "this trigger alias"
+        }`
+      : `Select all tasks in ${
+          selectedBuildVariants.length > 1 ? "these variants" : "this variant"
+        }`;
   const selectedTaskDisclaimerCopy = `${taskCount} task${
     taskCount !== 1 ? "s" : ""
   } across ${buildVariantCount} build variant${
     buildVariantCount !== 1 ? "s" : ""
-  }`;
+  }, ${aliasCount} trigger alias${aliasCount !== 1 ? "es" : ""}`;
```

Comment:
I don't know if I'm a fan of displaying  "X tasks across X build variants, X trigger aliases", because it will always be either tasks selected, or trigger aliases selected. I'd personally either only show "X trigger aliases" on the trigger aliases tabs and "X tasks across X build variants" on the main tab, or show "X tasks across X build variants" on the main tab but nothing on the trigger alias tab. I'm open to other opinions though, and I'm not against this being discussed/tackled in a separate ticket.